### ISSUE TITLE
Delvir0 - Rollovers could be blocked if emissionsToken is UDSC and a user has been blacklisted or if it's blocked on purpose

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-03-Y2K-judging#177]
Delvir0

high

# Rollovers could be blocked if emissionsToken is UDSC and a user has been blacklisted or if it's blocked on purpose

## Summary
Carousel.mintRollovers could become unavailable as it will fail if the emissonsToken is USDC is blacklisted
## Vulnerability Detail
USDC has a blacklist option which addresses can be added to. This means that transfer from and transfer to of USDC is prohibited for that address (user). 

Carousel.mintRollovers enters a while loop which goes over all indexes of `queue` "only if user won epoch he is rolling over" and "skip the rollover for the user if the assets cannot cover the relayer fee instead of revert" 
https://github.com/sherlock-audit/2023-03-Y2K/blob/main/Earthquake/src/v2/Carousel/Carousel.sol#L400-L406

After these two requirements emission tokens are transferred out of the contract so that users can access them.
https://github.com/sherlock-audit/2023-03-Y2K/blob/main/Earthquake/src/v2/Carousel/Carousel.sol#L414

Then, `emissionsTokens.safeTransfer` is called to send the emissionToken to the users, looping over `queue[index].receiver`.
If one of these `reveiver`'s is blacklisted by USDC and the emissonToken == USDC, the whole transaction will always revert at this point.

Assume Alice is not blacklisted by USDC and interacts with Y2K.
Alice uses the `enlistInRollover` function, assuming she is eligible and passes the requirements.
In the meantime, before `mintRollovers` is called, Alice does something malicious (according to CoinBase) and gets blacklisted.
`mintRollovers` gets called to the point where `emissionsToken.safeTransfer(queue[index].receiver,...)` is called and receiver = Alice.
Since Alice is blacklisted, transfers of USDC to Alice is not possible thus reverting the whole function making it unavailable.

While USDC is currently not the emissionsToken and Y2K is, it's possible that this is changed to USDC in the future.
This change would normally not trigger a need for an audit since it seems like a small change simply from on token to another and not changing any code logic.

NOTE: while this could be "unexpectedly" blocked by using USDC, it's also possible to make this function unavailable deliberately.
An user could create a malicious contract which fails to receive whatever `emissionsToken` is at `emissionsToken.safeTransfer(queue[index].receiver,...)`. 
## Impact
A core functionality of the project will not be available and emissions that should be sent to the users are stuck.
## Code Snippet
https://github.com/sherlock-audit/2023-03-Y2K/blob/main/Earthquake/src/v2/Carousel/Carousel.sol#L361
## Tool used

Manual Review

## Recommendation
Reconsider the way these funds are sent. Possible aspects to think about are:
- Incase of USDC, check if the user is blacklisted
- When sending funds in an array to multiple users, you'll need to make sure to have a solution implemented if a transactions fails "expectedly". In this case it would mean to continue anyway in the case of an failed transaction although this might also skip false negatives so a way to retrieve these tokens also needs to be available.

**Comment by Delvir0** [comment]
Escalate for 10 USDC
Duplicate of https://github.com/sherlock-audit/2023-03-Y2K-judging/issues/109 
*AND* 
https://github.com/sherlock-audit/2023-03-Y2K-judging/issues/497

Last line in Vulnerability Details I stated: 
"NOTE: while this could be "unexpectedly" blocked by using USDC, it's also possible to make this function unavailable deliberately.
An user could create a malicious contract which fails to receive whatever emissionsToken is at emissionsToken.safeTransfer(queue[index].receiver,...)." which is dup of https://github.com/sherlock-audit/2023-03-Y2K-judging/issues/497

**Comment by sherlock-admin** [comment]
> Escalate for 10 USDC
> Duplicate of https://github.com/sherlock-audit/2023-03-Y2K-judging/issues/109 
> *AND* 
> https://github.com/sherlock-audit/2023-03-Y2K-judging/issues/497
> 
> Last line in Vulnerability Details I stated: 
> "NOTE: while this could be "unexpectedly" blocked by using USDC, it's also possible to make this function unavailable deliberately.
> An user could create a malicious contract which fails to receive whatever emissionsToken is at emissionsToken.safeTransfer(queue[index].receiver,...)." which is dup of https://github.com/sherlock-audit/2023-03-Y2K-judging/issues/497

You've created a valid escalation for 10 USDC!

To remove the escalation from consideration: Delete your comment.

You may delete or edit your escalation comment anytime before the 48-hour escalation window closes. After that, the escalation becomes final.

**Comment by hrishibhat** [comment]
Escalation accepted

Valid duplicate of #109 
Since the primary issue in the submission is #109 considering the issue duplicate of the same. 
Requesting the Watson to submit issues separately in the future in case of valid issues.

**Comment by sherlock-admin** [comment]
> Escalation accepted
> 
> Valid duplicate of #109 
> Since the primary issue in the submission is #109 considering the issue duplicate of the same. 
> Requesting the Watson to submit issues separately in the future in case of valid issues. 
> 

This issue's escalations have been accepted!

Contestants' payouts and scores will be updated according to the changes made on this issue.
